neuraxle/metaopt/random.py

```diff
@@ -114,6 +117,278 @@ def _split(self, data_inputs):
+            data_inputs, expected_outputs)
+
+        train_data_inputs, train_expected_outputs = self.train_split(
+            data_inputs, expected_outputs)
+
+        return train_data_inputs, train_expected_outputs, validation_data_inputs, validation_expected_outputs
+
+    def train_split(self, data_inputs, expected_outputs=None) -> (List, List):
+        """
+        Split the data into train inputs, train expected outputs
+
+        Notes: The data supported by this cross validation is nd.array of shape = (batch, timeSeries, ...) .
+        The array can have an arbitrary number of dimension, but the time series axis is currently limited to `axis=1`.
+        Recommended format for the data is an numpy array of shape [batch_size, total_time_steps, n_features].
+
+        :param data_inputs: data to perform walk forward cross validation into.
+        :param expected_outputs: the expected target*label that will be used during walk forward cross validation.
+        :return: train_data_inputs, train_expected_outputs
+        """
+        splitted_data_inputs = self._train_split(data_inputs)
+        if expected_outputs is not None:
+            splitted_expected_outputs = self._train_split(expected_outputs)
+            return splitted_data_inputs, splitted_expected_outputs
+
+    def validation_split(self, data_inputs, expected_outputs=None) -> List:
+        """
+        Split the data into validation inputs, validation expected outputs.
+
+        Notes: The data supported by this cross validation is nd.array of shape = (batch, timeSeries, ...) .
+        The array can have an arbitrary number of dimension, but the time series axis is currently limited to `axis=1`.
+        Recommended format for the data is an numpy array of shape [batch_size, total_time_steps, n_features].
+
+        :param data_inputs: data to perform walk forward cross validation into.
+        :param expected_outputs: the expected target*label that will be used during walk forward cross validation.
+        :return: validation_data_inputs, validation_expected_outputs
+        """
+        splitted_data_inputs = self._validation_split(data_inputs)
+        if expected_outputs is not None:
+            splitted_expected_outputs = self._validation_split(expected_outputs)
+            return splitted_data_inputs, splitted_expected_outputs
+        return splitted_data_inputs
+
+    def _train_split(self, data_inputs):
+        splitted_data_inputs = []
+        number_step = self._get_number_fold(data_inputs)
+
+        for i in range(number_step):
+            # first slice index is always 0 for walk forward cross validation
+            a = int(i * self.validation_window_size)
+            b = int(a + self.training_window_size)
+
+            if b > data_inputs.shape[1]:
+                b = data_inputs.shape[1]
+
+            # TODO: The slicer could a inverse_transform of the joiner. A len method should also be defined.
```

Comment:
Interesting thought here!

Reply:
Perfect, could be implemented in another PR.

Reply:
Cool! 
Also see this issue which is related: https://github.com/Neuraxio/Neuraxle/issues/28